File: TypeScript/codewars/Grasshopper-Grade-book.ts
export function getGrade(a: number, b: number, c: number): string {
	const GRADE = {
		A: "A",
		B: "B",
		C: "C",
		D: "D",
		F: "F",
	};

	const avg = (a + b + c) / 3;

	if (avg >= 90) return GRADE.A;
	if (avg >= 80) return GRADE.B;
	if (avg >= 70) return GRADE.C;
	if (avg >= 60) return GRADE.D;
	return GRADE.F;
}

File: Csharp/CodeWars/src/Completed/GrasshopperGradeBook.cs
namespace CodeWars.Completed
{
    public class GrasshopperGradeBook
    {
        public static char GetGrade(int s1, int s2, int s3)
        {
            int avg = (int)new int[] { s1, s2, s3 }.Average();

            return avg switch
            {
                >= 90 and <= 100 => 'A',
                >= 80 and < 90 => 'B',
                >= 70 and < 80 => 'C',
                >= 60 and < 70 => 'D',
                < 60 => 'F',
                _ => throw new ArgumentOutOfRangeException(nameof(avg), "Invalid average score")
            };
        }
    }
}
